inliner panic on unexported symbol

There seems to be a bug in the inliner when compiling specs. Still trying to get enough details to reproduce, but user reported this stack trace: 

```
2022/09/07 13:22:26 compiling API rest to output versions
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 1 [running]:
reflect.valueInterface({0x100ba7340?, 0x140000f8448?, 0x100b42a9a?}, 0xc?)
        /Users/csnyk/sdk/go1.18/src/reflect/value.go:1441 +0xcc
reflect.Value.Interface(...)
        /Users/csnyk/sdk/go1.18/src/reflect/value.go:1430
github.com/snyk/vervet/v4.(*Inliner).Struct(0x14000112088, {0x100ba7340?, 0x140000f8448?, 0x1005f79cc?})
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/inliner.go:33 +0x48
github.com/mitchellh/reflectwalk.walkStruct({0x100ba7340?, 0x140000f8448?, 0x14000112088?}, {0x100b9c800, 0x14000112088})
        /Users/csnyk/go/pkg/mod/github.com/mitchellh/reflectwalk@v1.0.2/reflectwalk.go:370 +0xa8
github.com/mitchellh/reflectwalk.walk({0x100ba7340?, 0x140000f8448?, 0x14000112088?}, {0x100b9c800, 0x14000112088})
        /Users/csnyk/go/pkg/mod/github.com/mitchellh/reflectwalk@v1.0.2/reflectwalk.go:206 +0x5fc
github.com/mitchellh/reflectwalk.walkStruct({0x100c172e0?, 0x140000f8380?, 0x14000112088?}, {0x100b9c800, 0x14000112088})
        /Users/csnyk/go/pkg/mod/github.com/mitchellh/reflectwalk@v1.0.2/reflectwalk.go:404 +0x324
github.com/mitchellh/reflectwalk.walk({0x100c29020?, 0x140000f8380?, 0x14000112088?}, {0x100b9c800, 0x14000112088})
        /Users/csnyk/go/pkg/mod/github.com/mitchellh/reflectwalk@v1.0.2/reflectwalk.go:206 +0x5fc
github.com/mitchellh/reflectwalk.Walk({0x100c29020?, 0x140000f8380?}, {0x100b9c800, 0x14000112088})
        /Users/csnyk/go/pkg/mod/github.com/mitchellh/reflectwalk@v1.0.2/reflectwalk.go:99 +0x15c
github.com/snyk/vervet/v4.(*Inliner).Inline(...)
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/inliner.go:28
github.com/snyk/vervet/v4.(*Collator).mergeComponents(0x1400048dfb0, 0x140004bee60)
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/collator.go:221 +0x13c8
github.com/snyk/vervet/v4.(*Collator).Collate(0x1400048dfb0, 0x140004bee60)
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/collator.go:89 +0x300
github.com/snyk/vervet/v4.resourceVersionsSlice.at({0x140001123a0, 0x1, 0x140004f11e8?}, {{0x0, 0xeda98b280, 0x0}, 0x2})
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/resource_versions.go:58 +0xfc
github.com/snyk/vervet/v4.newSpecVersions({0x140001123a0, 0x1, 0x1})
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/spec.go:234 +0x504
github.com/snyk/vervet/v4.LoadSpecVersionsFileset({0x14000110a10, 0x1, 0x1400051f938?})
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/spec.go:64 +0x314
github.com/snyk/vervet/v4/internal/compiler.(*Compiler).Build(0x1400000ef00, {0x100795344?, 0x1400051f9f8?}, {0x1400031a25c, 0x4})
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/internal/compiler/compiler.go:279 +0x280
github.com/snyk/vervet/v4/internal/compiler.(*Compiler).apisEach(0x100c5e2f8?, {0x100c5e2f8, 0x14000032720}, 0x140004f1aa8)
        /Users/csnyk/go/pkg/mod/github.com/snyk/vervet/v4@v4.26.0/internal/compiler/compiler.go:248 +0xa4
```

--- a/inliner.go
+++ b/inliner.go
@@ -30,6 +30,9 @@ func (in *Inliner) Inline(doc *openapi3.T) error {
 
 // Struct implements reflectwalk.StructWalker
 func (in *Inliner) Struct(v reflect.Value) error {
+	if !v.CanInterface() {
+		return nil
+	}
 	switch val := v.Interface().(type) {
 	case openapi3.SchemaRef:
 		if _, ok := in.refs[val.Ref]; ok {
@@ -141,6 +144,9 @@ func (rr *RefRemover) RemoveRef() error {
 
 // Struct implements reflectwalk.StructWalker
 func (rr *RefRemover) Struct(v reflect.Value) error {
+	if !v.CanInterface() {
+		return nil
+	}
 	switch v.Interface().(type) {
 	case openapi3.SchemaRef:
 		valPointer := v.Addr().Interface().(*openapi3.SchemaRef)

--- a/remove_elements.go
+++ b/remove_elements.go
@@ -70,6 +70,9 @@ func (ex *excluder) apply() error {
 
 // Struct implements reflectwalk.StructWalker
 func (ex *excluder) Struct(v reflect.Value) error {
+	if !v.CanInterface() {
+		return nil
+	}
 	switch v.Interface().(type) {
 	case openapi3.ExtensionProps:
 		ex.applyExtensionProps(v.Addr().Interface().(*openapi3.ExtensionProps))

